tests/unit_tests/test_analytics/test_sentry.py

```diff
@@ -0,0 +1,586 @@
+        assert (
+            sentry_sdk.get_current_scope().client.dsn != wandb_sentry.scope.client.dsn  # type: ignore
+        )
+
+
+def test_wandb_error_reporting_disabled(relay: MetricRelayServer):
+    """
+    This test ensures no events are sent to sentry when the wandb sentry client is disabled.
+
+    The test sets the `ERROR_REPORTING` environment variable to `false` and initializes the wandb sentry client.
+    """
+    with mock.patch.dict(
+        os.environ,
+        {
+            wandb.env.ERROR_REPORTING: "false",
+            wandb.env.SENTRY_DSN: SENTRY_DSN_FORMAT.format(
+                key="WANDB_SENTRY_PUBLIC_KEY",
+                port=relay.port,
+                project="123456",
+            ),
+        },
+    ):
+        wandb_sentry = wandb.analytics.Sentry()
+        wandb_sentry.setup()
+        wandb_sentry.configure_scope(tags={"entity": "tag"})
+        wandb_sentry_event_id = wandb_sentry.message("wandb sentry message")
+
+        # When wandb sentry is disabled, the _safe_noop wrapper function always returns `None`
+        # Assert none is returned when the wandb Sentry client is disabled
+        assert wandb_sentry.scope is None
+        assert wandb_sentry_event_id is None
+
+
+def test_wandb_sentry_init_after_client_init(relay: MetricRelayServer):
+    """
+    This test ensures proper isolation between the Sentry instances initialized by another client and by wandb's client.
+
+    This test initializes the Sentry client first using `sentry_sdk.init()`, follwed by initializing wandb's Sentry client.
+    Both instances contain unique sessions and tags
+    """
+    expected_other_sentry_response = SentryResponse(
+        message="other sentry message",
+        project_id="654321",
+        public_key="OTHER_SENTRY_PUBLIC_KEY",
+        tags={"test": "tag"},
+    )
+    expected_wandb_sentry_response = SentryResponse(
+        message="wandb sentry message",
+        project_id="123456",
+        public_key="WANDB_SENTRY_PUBLIC_KEY",
+    )
+    with mock.patch.dict(
+        os.environ,
+        {
+            wandb.env.ERROR_REPORTING: "true",
+            wandb.env.SENTRY_DSN: SENTRY_DSN_FORMAT.format(
+                key=expected_wandb_sentry_response.public_key,
+                port=relay.port,
+                project=expected_wandb_sentry_response.project_id,
+            ),
+        },
+    ):
+        sentry_sdk.init(
+            dsn=SENTRY_DSN_FORMAT.format(
+                key=expected_other_sentry_response.public_key,
+                port=relay.port,
+                project=expected_other_sentry_response.project_id,
+            ),
+            default_integrations=False,
+        )
+        wandb_sentry = wandb.analytics.Sentry()
+        wandb_sentry.setup()
+
+        sentry_sdk.set_tag("test", "tag")
+        wandb_sentry.configure_scope(tags={"entity": "tag"})
+        expected_wandb_sentry_response.tags = wandb_sentry.scope._tags  # type: ignore
```

Comment:
why? i think you should do it when you initialize the expected response, don't reach for the internals?

Reply:
Thats fair, my initial thought was that some tags are auto added to sentry when calling `configure_scope` which at first I was more interested in testing that data (i.e. tags) between the 3p sentry client and the wandb sentry client is not duplicated.

now the tests have become more encompassing so it makes sense that we test that the tags we expect the `configure_scope` function to add are also present.